conan/tools/env/environment.py

```diff
@@ -520,12 +520,52 @@ def save_sh(self, file_location, generate_deactivate=True):
         content = f'script_folder="{os.path.abspath(filepath)}"\n' + content
         save(file_location, content)
 
+    def save_fish(self, file_location, generate_deactivate=True):
+        filepath, filename = os.path.split(file_location)
+        deactivate_file = os.path.join(filepath, "deactivate_{}".format(filename))
+        values = self._values.keys()
+        if len(values) == 0:
+            # Empty environment, nothing to restore (Easier to handle in Fish)
+            deactivate = ("""echo "echo Nothing to restore" > {deactivate_file}"""
+                          .format(deactivate_file=deactivate_file))
+        else:
+            deactivate = textwrap.dedent("""\
+               echo "echo Restoring environment" > "{deactivate_file}"
+               for v in {vars}
+                   set is_defined "true"
+                   set value (printenv $v); or set is_defined ""
+                   if test -n "$value" -o -n "$is_defined"
+                       echo set -gx "$v" "$value" >> "{deactivate_file}"
+                   else
+                       echo set -ge "$v" >> "{deactivate_file}"
+                   end
+               end
+               """.format(deactivate_file=deactivate_file, vars=" ".join(self._values.keys())))
+        capture = textwrap.dedent("""\
+              {deactivate}
+              """).format(deactivate=deactivate if generate_deactivate else "")
+        result = [capture]
+        for varname, varvalues in self._values.items():
+            value = varvalues.get_str("${name}", self._subsystem, pathsep=self._pathsep)
+            value = value.replace('"', '\\"')
+            if value:
+                result.append('set -gx {} "{}"'.format(varname, value))
+            else:
+                result.append('set -ge {}'.format(varname))
+
+        content = "\n".join(result)
+        content = relativize_generated_file(content, self._conanfile, "$script_folder")
```

Comment:
Isn't it possible to not use this and relativize paths as the other environment do?